### PR TITLE
Patch

### PR DESCRIPTION
**Pull request by hotoo**
* optimize regexp.
* refact argument to array.

### CODE REVIEW

**Comment by popomore** [comment]
@fengmk2

**Comment by popomore** [comment]
+1

File: index.js [hunk 1]
```diff
@@ -35,7 +35,7 @@ module.exports = function (app, options) {
     var names = fs.readdirSync(localeDir);
     for (var i = 0; i < names.length; i++) {
       var name = names[i];
-      if (!/\.(js|json|properties)$/.test(name)) {
+      if (!/\.(?:js|json|properties)$/.test(name)) {
```
Comment (fengmk2) [inline index.js]: `test()` won't catch any match items.

File: index.js [hunk 2]
```diff
@@ -109,11 +109,8 @@ module.exports = function (app, options) {
     }
 
     // __(key, value1, value2, value3, value4, value5, ...)
-    var args = new Array(arguments.length);
+    var args = Array.prototype.slice.call(arguments);
     args[0] = text;
-    for(var i = 1; i < args.length; i++) {
-      args[i] = arguments[i];
-    }
```
Comment (fengmk2) [inline index.js]: for loop will more faster than slice...

image

File: index.js (hunk 1 above)
Comment (hotoo) [inline index.js]: NO. catch group will catch it, whether `test` or `exec`, catch group can be used in regexp self, like `/(["']).*?\1/`. `\1` is reference the catch group.

**Pull request merged**